internal/workflow.go

```diff
@@ -692,25 +690,24 @@ func (wc *workflowEnvironmentInterceptor) ExecuteChildWorkflow(ctx Context, chil
 		}
 	}, func(r WorkflowExecution, e error) {
 		if e == nil {
-			childWorkflowExecution = &r
+			if cancellable {
+				cancellationCallback.fn = func(v interface{}, _ bool) bool {
+					if ctx.Err() == ErrCanceled && !mainFuture.IsReady() {
+						// child workflow started, and ctx cancelled
+						getWorkflowEnvironment(ctx).RequestCancelChildWorkflow(options.Namespace, r.ID)
+					}
+					return false
+				}
+				_, ok, more := ctxDone.receiveAsyncImpl(cancellationCallback)
+				if ok || !more {
+					cancellationCallback.fn(nil, more)
+				}
+			}
 		}
+
 		executionSettable.Set(r, e)
 	})
 
-	if cancellable {
```

Comment:
This was the bug:

1) Workflow start command would be sent, but not yet finished.
2) Workflow would be cancelled
3) CancellationCallback.fn would be invoked. But because the Workflow hasn't started yet, childWorkflowExection was equal to null.
4) As a result, we did not request a cancellation of the child workflow.
5) Parent workflow is stuck "running" because the child is never requested to cancel this.